### DISCUSSION TITLE
feat: Support trusted name header for user signup

### DISCUSSION THREAD

**Discussion by sample** [open-webui/open-webui#2864] [General]
Hello everyone.

I propose adding the `WEBUI_AUTH_TRUSTED_NAME_HEADER` parameter, so that during registration via `WEBUI_AUTH_TRUSTED_EMAIL_HEADER`, the username would also be set.
If the header is not present, the email would be used as before.
Implementation in PR #2706

**Comment by codingstube** [comment]
Hello everyone,
I have set up an Apache2 reverse proxy for our test environment, which performs authentication via Kerberos. 
The HTTP headers are then filled:
- WEBUI_AUTH_TRUSTED_EMAIL_HEADER=X-User-Email
- WEBUI_AUTH_TRUSTED_NAME_HEADER=X-User-Givenname


Enclosed is the Apache configuration. There may be tips for optimizing it. 
What I have not managed to do is to make an LDAP query after the Kerberos authentication, which fills in the first name of the user. I have only entered the user name as the domain name. 
All LDAP queries were accepted, but the LDAP environment variables remained null.
Thanks!


```
<VirtualHost *:80>
  <Location /api/v1/auths/signin >

    ##
    AddOutputFilterByType DEFLATE text/html
Options FollowSymLinks SymLinksIfOwnerMatch
    LogLevel debug
    AuthType                   Kerberos
    AuthName                   "Web Application"
    KrbAuthRealms              DOMAIN.COM
    Krb5Keytab                 /etc/key.keytab
    KrbMethodNegotiate         on
    KrbAuthoritative           on
    KrbMethodK5Passwd          on
    KrbLocalUserMapping        on
    KrbSaveCredentials         on
    require valid-user

    RewriteEngine On
    RewriteCond %{LA-U:REMOTE_USER} (.+)
    RewriteRule .* - [E=RU:%1]
RequestHeader add X-User-Email %{RU}e@domain.com

   RequestHeader set X-User-Givenname "%{RU}e"
</Location>

ProxyTimeout 600
KeepAlive On
  ProxyPreserveHost ON

  ProxyPass / http://open-webui:3000/
  ProxyPassReverse / http://open-webui:3000/


  ProxyPass "/ws/socket.io/" "wss://open-webui:3000/ws/socket.io/"
  ProxyPassReverse "/ws/socket.io/" "wss://open-webui:3000/ws/socket.io/"
LogFormat "%h %l %u %t \"%r\" %>s %b \"%{User-Agent}i\" \"%{X-LDAP-Var}i\"" ldap
CustomLog /var/log/apache2/ldap.log ldap


CustomLog /var/log/apache2/env_vars_log "%{RU}e"
CustomLog /var/log/apache2/env_vars_log "%{ENV:AUTHENTICATE_GIVENNAME}e"
</VirtualHost>
```
